### ISSUE TITLE
Multi-Account-Integration

### ISSUE THREAD

**Issue by maste9** [SchildiChat/SchildiChat-android#21]
Are there already plans to add support for multiple accounts? I have more than one Matrix-Account and would appreciate to watch them without having to log out and in again.

Best regards!

**Comment by SpiritCroc** [comment]
No, this is not planned, as it would likely require way more work than I want to sacrifice for this project.
For my personal use, I'm building [multiple Schildis](https://matrix-client.matrix.org/_matrix/media/r0/download/matrix.org/uwGtOOEtFBIOSPPrVMtLRzyY) that can be installed next to each other.
These are not publicly released at the moment (referring to the apks; [the source code is available](https://github.com/SpiritCroc/SchildiChat-android/blob/sc/alternative_package.sh)), though, but if there is demand, maybe I can add one or two to my F-Droid repo.

**Comment by samuel-p** [comment]
There is an issue for this at Element Android: https://github.com/vector-im/element-android/issues/26

So maybe this is an upstream issue instead of wontfix?

**Comment by SpiritCroc** [comment]
I assigned the "wontfix" label because I personally won't work on it. I guess the label is misleading, I'll change it.
If upstream (or somebody else?) comes up with a good implementation, I can include it in Schildi as well, of course.

**Comment by serrq** [comment]
Having two or more Schilds installed means a greater consumption of hardware and battery resources. I will definitely not follow this path.

I hope someone has the computer skills and goodwill to implement the multi account in the code.

**Comment by SpiritCroc** [comment]
Since element-android will be deprecated in favour of element-x, it probably doesn't make much sense to build this kind of bigger feature into this codebase. Let's hope it'll be prioritized a bit more for element-x.

(I'm not 100% sure about the future of SchildiChat-android in that matter, but most likely I'll rebuild it in some form or another on top of element-x as well... once it's beta enough to work with)

**Comment by q-TheDog** [comment]
For those that just need a way to access basic messaging features for your alternate accounts on mobile without installing other apps and background services, I personally recommend using [Hydrogen Web](https://hydrogen.element.io/). It supports multi-account (albeit only letting you look at and sync one at a time).

Also, just as another idea related to SpiritCroc installing multiple copies of SchilidiChat: One could probably also use something like [Revanced Manager](https://github.com/ReVanced/revanced-manager) to easily install multiple copies of Schilidi (it lets you patch any app, not just YouTube, and comes with changing the app ID/icon for loading multiple copies). Just one more way of doing it all within Android.

**Comment by Coehill** [comment]
Personally this is my most needed feature and whichever Android client has it first will become my sole Matrix app.

**Comment by Coehill** [comment]
> Since element-android will be deprecated in favour of element-x, it probably doesn't make much sense to build this kind of bigger feature into this codebase. Let's hope it'll be prioritized a bit more for element-x.
> 
> (I'm not 100% sure about the future of SchildiChat-android in that matter, but most likely I'll rebuild it in some form or another on top of element-x as well... once it's beta enough to work with)

This is not prioritized im Element-x unfortunately.
https://github.com/element-hq/element-android/issues/26

**Comment by SpiritCroc** [comment]
> Personally this is my most needed feature and whichever Android client has it first will become my sole Matrix app.

FluffyChat had it last time I checked

**Comment by maste9** [comment]
> > Personally this is my most needed feature and whichever Android client has it first will become my sole Matrix app.
> 
> FluffyChat had it last time I checked

It does. It is still marked as beta btw.. If you like being noticed about new messages that is up to now only working for the first account. Every other account has to be checked by hand. Hopefully that will be improved in future versions.

**Comment by dreirund** [comment]
> FluffyChat had it last time I checked

Cinny, too.
